python/cudf_polars/cudf_polars/experimental/io.py

```diff
@@ -127,9 +127,21 @@ def from_scan(
             column_stats = stats.column_stats.get(ir, {})
             column_sizes: list[int] = []
             for cs in column_stats.values():
-                storage_size = cs.source_info.storage_size
-                if storage_size.value is not None:
-                    column_sizes.append(storage_size.value)
+                element_size = 1
+                if (dtype := ir.schema.get(cs.name)) is not None:
+                    if dtype.id() in (
+                        plc.TypeId.STRING,
+                        plc.TypeId.FLOAT64,
+                        plc.TypeId.INT64,
+                    ):
+                        element_size = 8
+                    elif dtype.id() in (plc.TypeId.FLOAT32, plc.TypeId.INT32):
+                        element_size = 4
```

Comment:
Not sure if we have this problem for dtypes besides String. I probably need to explore and add a more-detailed test for this.

Reply:
Mmm I see: the problem right now is that we sometimes underestimate the in-memory size, and so we're adding a lower bound on the element size.

I do expect this would affect something like a `Struct` with string fields, since those string fields should be dictionary encoded as well.

Reply:
Decided to pull out all this logic and just use a simple lower bound of "1 byte per row" for now.